### ISSUE TITLE
SSL match_hostname does not accept IP Address

### ISSUE THREAD

**Issue by 06b739dd-9b4a-470d-9d2b-d567a22ca7a6** [python/cpython#74375]
BPO | [30189](https://bugs.python.org/issue30189)
--- | :---
Nosy | @tiran, @emeve89

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-04-27.18:32:07.617>
created_at = <Date 2017-04-27.18:16:08.509>
labels = ['extension-modules', 'type-feature']
title = 'SSL match_hostname does not accept IP Address'
updated_at = <Date 2017-04-27.18:32:07.615>
user = 'https://github.com/emeve89'
```

bugs.python.org fields:
```python
activity = <Date 2017-04-27.18:32:07.615>
actor = 'christian.heimes'
assignee = 'none'
closed = True
closed_date = <Date 2017-04-27.18:32:07.617>
closer = 'christian.heimes'
components = ['Extension Modules']
creation = <Date 2017-04-27.18:16:08.509>
creator = 'emeve89'
dependencies = []
files = []
hgrepos = []
issue_num = 30189
keywords = []
message_count = 2.0
messages = ['292468', '292470']
nosy_count = 2.0
nosy_names = ['christian.heimes', 'emeve89']
pr_nums = []
priority = 'normal'
resolution = 'wont fix'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue30189'
versions = ['Python 2.7']
```

</p></details>

**Comment by 06b739dd-9b4a-470d-9d2b-d567a22ca7a6** [comment]
I need this patch to be available in python 2.7.13
https://bugs.python.org/issue23239

**Comment by tiran** [comment]
In general we don't backport features. https://www.python.org/dev/peps/pep-0466/ allows backports of network security enhancements. However IP address validation is not a security enhancement. You can either update to a new version of Python or role your own verification code.
